core/common/src/main/java/alluxio/AbstractClient.java

```diff
@@ -194,18 +194,13 @@ public synchronized void connect() throws IOException, ConnectionFailedException
    * if the client has not connected with the remote for a while, for example.
    */
   public synchronized void disconnect() {
-    if (mConnected) {
+    beforeDisconnect();
```

Comment:
does it makes sense to call this and `afterDisconnect` even if `mConnected` is already false?